trench/backends/basic_mail.py

```diff
@@ -39,3 +42,9 @@ def dispatch_message(self) -> DispatchResponse:
         except ConnectionRefusedError as cause:  # pragma: nocover
             logging.error(cause, exc_info=True)  # pragma: nocover
             return FailedDispatchResponse(details=str(cause))  # pragma: nocover
+
+
+class SendMailHotpMessageDispatcher(
+    AbstractHotpMessageDispatcher, SendMailMessageDispatcher
```

Comment:
For the mail, SMS, and Twilio shouldn't this be the default (`SendMailMessageDispatcher` should extend `AbstractHotpMessageDispatcher`)?

My project is having other issue with SMS Twilio in that often times codes are not valid because generates a new TOTP code from generate/validate and when the validate calls a new MFA code could have been generated after the message was sent. I was looking at other issues for the project and could not find any however can reproduce. I like this solution and makes more sense for these MFA backends.

Reply:
I agree this behavior is typically standard for sms based codes. I did not change it to be the default as that would be a breaking change for existing trench users using the default settings.

On the other issue, can you remedy a new code being generated by extending the validity window?